airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/job_tracker/TrackingMetadata.java

```diff
@@ -102,9 +110,38 @@
             metadata.put("volume_rows", syncSummary.getRecordsSynced());
           }
         }
+
+        final List<FailureReason> failureReasons = failureReasonsList(attempts);
+        if (!failureReasons.isEmpty()) {
+          metadata.put("failure_reasons", failureReasonsListAsJson(failureReasons).toString());
+          metadata.put("main_failure_reason", failureReasonAsJson(failureReasons.get(0)).toString());
```

Comment:
We sort failureReasons chronologically in the `FailureHelper` class, so getting index 0 should be pretty reliable here, but it might make sense to explicitly sort here so that we're not relying on the assumption that these are sorted how the job tracker expects